### ISSUE TITLE
Slight inaccuracy in manual for bottom qualifier and `null`?

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#3329]
I was reading the manual documentation around qualifier hierarchies and the bottom qualifier:

https://checkerframework.org/manual/#creating-bottom-qualifier

It states that `null` should always have the bottom type.  But in the returns receiver checker, `null` has the top type.  This is because we want the following code to _fail_ to type check:

```java
class Foo {
  @This Foo m(@This Foo this) {
    return null;
  }
}
```

`@This` is a polymorphic qualifier intended to force a method to return its receiver when it appears on the return type.  If `null` had the bottom type in that type system, the above code would type check, which is not what is desired.

It's possible that the Returns Receiver checker is just strange, and it's best to leave this case out of the manual.  Just wanted to mention it since we now have a counterexample to what is written in the manual.

**Comment by wmdietl** [comment]
The manual already mentions the Nullness Checker as a type system that doesn't follow that guideline: `(The only exception is if the type system has special treatment for null values, as the Nullness Checker does. In that case, add the meta-annotation @QualifierForLiterals(LiteralKind.NULL) to the correct qualifier.)`

So it is a guideline that is generally true, but there can be "strange" type systems that have other requirements.

**Comment by msridhar** [comment]
Wow, don't know how I missed that 🤦‍♂️Sorry for the noise!
